packages/discovery-react-components/src/components/SearchFacets/SearchFacets.tsx

```diff
@@ -101,19 +109,29 @@ const SearchFacets: FC<SearchFacetsProps> = ({
         label: collection.name || ''
       };
     });
+
   const [collectionSelectionState, setCollectionSelectionState] = useState<
     SelectedCollectionItems['selectedItems']
   >(initialSelectedCollections);
+
+  const [isError, setIsError] = useState(false);
+
   const { fetchAggregations, performSearch } = useContext(SearchApi);
   const aggregations = aggregationResults || [];
   const mergedMessages = { ...defaultMessages, ...messages };
+
   const componentSettingsAggregations =
     overrideComponentSettingsAggregations ||
     (componentSettings && componentSettings.aggregations) ||
     [];
 
   useEffect(() => {
-    fetchAggregations(searchParameters);
+    setIsError(false);
+    try {
+      fetchAggregations(searchParameters);
+    } catch (error) {
+      setIsError(true);
+    }
```

Comment:
What's the best way to make the fetch run again, in case component is re-rendered? But in such a way that doesn't send out a ton of requests (sometimes, these components are rendered a lot, even if nothing actually changes in the DOM).